intercepting-filter/src/main/java/com/iluwatar/intercepting/filter/Client.java

```diff
@@ -37,20 +35,20 @@
 import javax.swing.SwingUtilities;
 
 /**
- * The Client class is responsible for handling the input and running them through filters inside
- * the {@link FilterManager}.
+ * The Client class is responsible for handling the input and running them through filters inside the
+ * {@link FilterManager}.
  *
- * This is where {@link Filter}s come to play as the client pre-processes the request before being
- * displayed in the {@link Target}.
+ * This is where {@link Filter}s come to play as the client pre-processes the request before being displayed in the
+ * {@link Target}.
  * 
  * @author joshzambales
  *
  */
-public class Client extends JFrame {
+public class Client extends JFrame { // NOSONAR
 
   private static final long serialVersionUID = 1L;
 
-  private FilterManager filterManager;
+  private transient FilterManager filterManager;
```

Comment:
It did not introduce any problem. 
Alternate solution would be making FilterManager and its parents as Serializable. In this example, we are not transferring Client instance or planning to serialize, So I decided to make it transient to keep it simple. 

Let me know if you prefer marking all classes as Serializable as long as it does not have any non-serialized attributes.